File: 101-109_small_problems/easy_6/delete_vowels.rb
# Delete vowels
# Write a method that takes an array of strings, and returns an array of the same string values, except with the vowels (a, e, i, o, u) removed.

# Input: An array of strings
# Output: An array of strings
# Mental model of problem: Given an array of strings, create a new array containing the same strings with the vowels removed.

# Example:

# remove_vowels(%w(abcdefghijklmnopqrstuvwxyz)) == %w(bcdfghjklmnpqrstvwxyz)
# remove_vowels(%w(green YELLOW black white)) == %w(grn YLLW blck wht)
# remove_vowels(%w(ABC AEIOU XYZ)) == ['BC', '', 'XYZ']

# Data Structures

# Input and output are both arrays.

# Algorithms

# Get input array
# Create new array without_vowels
# Loop through input array
#   Remove vowels from current element
#   Add current element to without_vowels

# Return without_vowels

# Code
def remove_vowels(ary)
 without_vowels = []

 ary.each do |word|
   without_vowels << word.gsub(/[aeiou]/i, "")
 end

 without_vowels
end


p remove_vowels(%w(abcdefghijklmnopqrstuvwxyz)) == %w(bcdfghjklmnpqrstvwxyz)
p remove_vowels(%w(green YELLOW black white)) == %w(grn YLLW blck wht)
p remove_vowels(%w(ABC AEIOU XYZ)) == ['BC', '', 'XYZ']

# Further Exploration
# Ruby has all sorts of String methods that could accomplish this task. How did you end up solving this exercise?

File: exercises/js210_small_problems/string_and_text/deleteVowels.js
// Delete Vowels
// Write a function that takes an array of strings, and returns an array of the same strings values without the vowels (a, e, i, o, u).

// Input: An array of strings
// Output: An array of strings without the vowels

// Rules:
// - output a new array
// - output array contains same strings as the input, but without the vowels

// transformation - return an array with the same number of elements

// - remember that for the regex, that it has to account for both cases

// Data Structure:

// an array for both input an output

// Algorithm
// 1. Loop through the array using map
// 2.   Replace vowels in the string with ''
// 3. Return the transformed array

// Code

function removeVowels(array) {
  return array.map((string) => string.replace(/[aeiou]/gi, ''));
}


// Examples:

console.log(removeVowels(['abcdefghijklmnopqrstuvwxyz']));         // ["bcdfghjklmnpqrstvwxyz"]
console.log(removeVowels(['green', 'YELLOW', 'black', 'white']));  // ["grn", "YLLW", "blck", "wht"]
console.log(removeVowels(['ABC', 'AEIOU', 'XYZ']));                // ["BC", "", "XYZ"]
